.circleci/config.yml

```diff
@@ -208,7 +208,7 @@ workflows:
           OPENSSL_PREINSTALL: openssl
       - macOS:
           name: macOS-shared
-          CMAKE_ARGS: -DBUILD_SHARED_LIBS=ON -DOQS_DIST_BUILD=OFF -DOQS_ENABLE_KEM_CLASSIC_MCELIECE=OFF
+          CMAKE_ARGS: -DOQS_DIST_BUILD=OFF -DOQS_ENABLE_KEM_CLASSIC_MCELIECE=OFF
```

Comment:
If we remove `-DBUILD_SHARED_LIBS=ON` here, we ought to add the new `-DOQS_PROVIDER_BUILD_STATIC=ON` somewhere (else): Did I overlook such CI test case?